lib/masamune/thor.rb

```diff
@@ -162,16 +162,35 @@ def top_level?
           end
 
           def invoke_command(command, *args)
-            environment.with_exclusive_lock(command.name, non_blocking: true) do
```

Comment:
issue was that `command.name` sometimes returns a non fully qualified name, e.g. without a namespace. this causes issues if 2 tasks are named the same but in different namespaces- especially if they are invoked in parallel